pandas/core/series.py

```diff
@@ -2519,12 +2528,13 @@ def _binop(self, other, func, level=None, fill_value=None):
         with np.errstate(all='ignore'):
             result = func(this_vals, other_vals)
         name = ops.get_op_result_name(self, other)
-        result = self._constructor(result, index=new_index, name=name)
-        result = result.__finalize__(self)
-        if name is None:
-            # When name is None, __finalize__ overwrites current name
-            result.name = None
-        return result
+        if isinstance(result, tuple):
```

Comment:
Hopefully replace with `_construct_result` and `_construct_divmod_result`.

Reply:
Thanks for reminding me that two utilities. After `_construct_result` or `_construct_divmod_result` we still need to invoke `__finalize__` to propagate metadata and overwrite `name` again. Any suggestion on implementing such construction in a more concise way ?

Reply:
in core/ops.py, we call `res_name = get_op_result_name(left, right)`. Is that an option here?

Reply:
However after `_construct_(divmod_)result`, we still need `__finalize__`, which may overwrite `name`. Thus after `__finalize__` we need assign `name` again.

Reply:
Or should we do `__finalize__` in `_construct_result` in core/ops.py ?

Reply:
I think it's ok to do `__finalize__ ` in `_construct_result` since `_construct_result` is only used in core/ops.py at where the `__finalize__ ` should be called (but not now).

Will do that.